### DISCUSSION TITLE
What is the best way to select a computed property?

### DISCUSSION THREAD

**Discussion by bill-o-more** [mikro-orm/mikro-orm#4662] [Q&A] [answered]
I want to get the following example query: `SELECT "name", levenshtein(name, 'To kill a mockignbrid') as distance FROM book`.

My model looks like this:
```typescript
@Entity()
export class Book  {
    @PrimaryKey({ columnType: 'uuid', defaultRaw: `uuid_generate_v4()` })
    id = uuidv4();

    @Property()
    name: string;

    @Property()
    pages: number;

    @Property({ nullable: true, persist: false })
    distance?: number;
}
``` 

I've tried to
```typescript
const books =  await em.qb(Book, 'b1')
    .select(['id', 'name', `levenshtein(b1.name, 'To kill a mockignbrid') as distance`])
    .limit(10);
```
and got an array of `Book` objects withouth the distance property at all,

and
```typescript
const books =  await em.qb(Book, 'b1')
    .select(['id', 'name'])
    .select(`levenshtein(b1.name, 'To kill a mockignbrid')`).as('distance')
    .limit(10);
```
and got  an array of objects like `{ levenshtein: 7 }` - even ignoring the alias.

Is subqueries is my only way to go here? Can't find something like this in [docs](https://mikro-orm.io/docs/query-builder) that doesn't use a subquery, but making a subquery from the same table doesn't feel right...

**Comment by B4nan** [comment, marked as answer]
Can you prepare a complete repro? The first variant should work, we have similar tests to that. 

Note that the problem is entity mapping probably, you can always use `qb.execute()` which will give you raw results.

> and got an array of objects like { levenshtein: 7 } - even ignoring the alias.

The `qb.as()` method is not for aliasing fields, but for aliasing subqueries, the second try is indeed wrong. You are not aliasing the expression in select, you are aliasing the whole subquery you just created with the `as()` call (and awaiting that, which is actually an instance of knex QB, so no entity mapping at all).

**Comment by bill-o-more** [comment]
well, that was not entity mapping, but a postgres behavior I didn't know about, as I'm pretty new to it.
in my actual model, the property name was `levenshteinDistance`, and it turns out that if I don't quote the alias in the `as` clause, postgres does a lowercase on it. This resulted in a column named `levenshteindistance`, which did't map to anything in the entity of course. When I quoted the `as levenshteinDistance` it worked, but what I actually needed to remember is that in DB everything is snake-cased - so it should have been `levenshtein_distance` (which worked as well, left the quotes for consistency). Thanks!
